routers/web/repo/commit.go

```diff
@@ -288,13 +286,26 @@ func Diff(ctx *context.Context) {
 		commitID = commit.ID.String()
 	}
 
-	diff, err := gitdiff.GetDiffCommitWithWhitespaceBehavior(gitRepo,
-		commitID, ctx.FormString("skip-to"), setting.Git.MaxGitDiffLines,
-		setting.Git.MaxGitDiffLineCharacters, setting.Git.MaxGitDiffFiles,
-		gitdiff.GetWhitespaceFlag(ctx.Data["WhitespaceBehavior"].(string)),
-		false)
+	fileOnly := ctx.FormBool("file-only")
+	maxLines, maxLineCharacters, maxFiles := setting.Git.MaxGitDiffLines, setting.Git.MaxGitDiffLineCharacters, setting.Git.MaxGitDiffFiles
+	files := ctx.FormStrings("files")
+	if fileOnly && (len(files) == 2 || len(files) == 1) {
+		maxLines, maxLineCharacters, maxFiles = -1, 4096, -1
+		if setting.Git.MaxGitDiffLineCharacters > maxLineCharacters {
+			maxLineCharacters = setting.Git.MaxGitDiffLineCharacters
```

Comment:
Interesting, why is their a minimum of 4096 being enforced?

Reply:
This is when a user has specifically requested to review a long file. I will remove this and we can reconsider my point about long lines in the related issue.